core/app/[locale]/(default)/compare/_components/add-to-cart/index.tsx

```diff
@@ -3,66 +3,72 @@
 import { FragmentOf } from 'gql.tada';
 import { AlertCircle, Check } from 'lucide-react';
 import { useTranslations } from 'next-intl';
-import { useFormStatus } from 'react-dom';
+import { useTransition } from 'react';
 import { toast } from 'react-hot-toast';
 
 import { AddToCartButton } from '~/components/add-to-cart-button';
+import { useCart } from '~/components/header/cart-provider';
 import { Link } from '~/components/link';
 
 import { addToCart } from '../../_actions/add-to-cart';
 
 import { AddToCartFragment } from './fragment';
 
-const Submit = ({ data: product }: { data: FragmentOf<typeof AddToCartFragment> }) => {
-  const { pending } = useFormStatus();
-
-  return <AddToCartButton data={product} loading={pending} />;
-};
-
 export const AddToCart = ({ data: product }: { data: FragmentOf<typeof AddToCartFragment> }) => {
   const t = useTranslations('Compare.AddToCart');
+  const cart = useCart();
+  const [isPending, startTransition] = useTransition();
 
-  return (
-    <form
-      action={async (formData: FormData) => {
-        const result = await addToCart(formData);
-        const quantity = Number(formData.get('quantity'));
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const formData = new FormData(event.currentTarget);
+    const quantity = Number(formData.get('quantity'));
 
-        if (result.error) {
-          toast.error(t('error'), {
-            icon: <AlertCircle className="text-error-secondary" />,
-          });
+    // Optimistic update
+    cart.increment(quantity);
```

Comment:
🍹 Just thinking it might be nice to explicitly mention it's an optimistic update with the fn name.
```suggestion
    cart.optimisticlyIncrement(quantity);
```